cmd/gen-crl/main.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path"
+	"time"
+
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/pkcs11helpers"
+)
+
+type crlDef struct {
+	num   int
+	start time.Time
+	end   time.Time
+}
+
+func generateDefs(clk clock.Clock, total, initialNum int, startStr string, period, overlap time.Duration) ([]crlDef, error) {
+	if overlap >= period {
+		return nil, errors.New("CRL overlap must be smaller than the validity period")
+	}
+	var crlDefs []crlDef
+	start := clk.Now()
+	if startStr != "" {
+		var err error
+		start, err = time.Parse(time.RFC3339, startStr)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse start-date %q: %s", startStr, err)
+		}
+	}
+	end := start.Add(period)
+	for i := 0; i < total; i++ {
+		crlDefs = append(crlDefs, crlDef{
+			num:   i + initialNum,
+			start: start,
+			end:   end,
+		})
+		start = end.Add(-overlap)
+		end = start.Add(period)
+		log.Printf("CRL %d %s - %s\n", crlDefs[i].num, crlDefs[i].start, crlDefs[i].end)
+	}
+	return crlDefs, nil
+}
+
+func generateCRL(issuer *x509.Certificate, privKey crypto.Signer, def crlDef, dir, prefix string) error {
+	crl, err := issuer.CreateCRL(rand.Reader, privKey, nil, def.start, def.end)
+	if err != nil {
+		return fmt.Errorf("failed to create CRL %d: %s", def.num, err)
```

Comment:
Just to be paranoid, let's open the target file for writing before we call CreateCRL. That way if there's a permission denied error, we fail out before doing any signing.